core/app/models/spree/product.rb

```diff
@@ -280,6 +280,19 @@ def find_variant_property_rule(option_value_ids)
       end
     end
 
+    # An enumerable of media that goes along with a product
+    # Stores may not necessarily want to attach images to products (variants)
+    # This provides a more robust interface
+    #
+    # @return [Enumerable] of media for a product (and its variants)
+    def gallery
+      if Product.method_defined?(:variant_images)
```

Comment:
We define this association, so we shouldn't test for it. If this is to prepare for a future paperclip removal PR, it should be done later.